pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go

```diff
@@ -179,8 +182,10 @@ func applyLocalityFailover(
 				}
 			}
 		}
-		loadAssignment.Endpoints[i].Priority = uint32(priority)
-		priorityMap[priority] = append(priorityMap[priority], i)
+
+		priorityInt := int(loadAssignment.Endpoints[i].Priority*5) + priority
```

Comment:
prefer add some comments for others to understand

Reply:
PTAL